netbox/extras/scripts.py

```diff
@@ -446,6 +510,44 @@ def load_json(self, filename):
 
         return data
 
+    def run_test_scripts(self, job):
+        """
+        Run the report and save its results. Each test method will be executed in order.
+        """
+        self.logger.info(f"Running report")
+
+        # Perform any post-run tasks
+        self.pre_run()
```

Comment:
Let's move `pre_run()` and `post_run()` out of this method and into `run()`, to better separate functions.